pkg/fqdn/cache.go

```diff
@@ -595,6 +595,18 @@ func (c *DNSCache) ForceExpire(expireLookupsBefore time.Time, nameMatch *regexp.
 	return slices.Unique(namesAffected)
 }
 
+// Names returns the FQDN entries in the cache
+func (c *DNSCache) Names() (names []string) {
+	c.Lock()
+	defer c.Unlock()
```

Comment:
```suggestion
	c.RLock()
	defer c.RUnlock()
```
should suffice I think.

Reply:
Good point, fixed.